### PR TITLE
HLRC: Add Get Lifecycle Policy API to HLRC

### PR DESCRIPTION
**Pull request by AthenaEryma**
Adds Request and Reponse classes for accessing lifecycle policies.
Changes existing tests to use these classes where appropriate.
Sets up SPI configuration to allow parsing *Actions from XContent.

Relates to https://github.com/elastic/elasticsearch/issues/33100

### CODE REVIEW

**Review by colings86 — APPROVED** [review]
LGTM but maybe @hub-cap should also have a look before we merge since this adds some SPI stuff I'm not so familiar with

**Comment by hub-cap** [comment]
+1 to using SPI for the custom XContent. cc @talevy who was also asking me about this the other day.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/indexlifecycle/GetLifecyclePolicyResponse.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    public GetLifecyclePolicyResponse(ImmutableOpenMap<String, LifecyclePolicy> policies) {
+        this.policies = policies;
+    }
+
+    public ImmutableOpenMap<String, LifecyclePolicy> getPolicies() {
+        return policies;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/indexlifecycle/GetLifecyclePolicyResponse.java]: Do you need this? I dont see it being used anywhere.

**Review by hub-cap — CHANGES_REQUESTED** [review]
I think it would make sense to create a ILM SPI provider instead of a generic one, or else I might end up splitting it anyway :D

File: client/rest-high-level/src/main/java/org/elasticsearch/client/indexlifecycle/GetLifecyclePolicyResponse.java (hunk 1 above)
Comment (gwbrown) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/indexlifecycle/GetLifecyclePolicyResponse.java]: It's used to leverage `AbstractXContentTestCase` for testing parsing.  If there's a better way of doing that, please let me know!

Comment (hub-cap) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/indexlifecycle/GetLifecyclePolicyResponse.java]: I dont have a good answer for this yet. While we can totally test this, the value in the server -> client.fromXContent is greater than just testing the client.toXContent -> client.fromXContent. I think these kinds of tests are not really providing much value, and we also test the former in the IT tests.

**Comment by gwbrown** [comment]
@elasticmachine retest this please

The failure does not reproduce locally.

**Review by hub-cap — APPROVED** [review]

**Pull request merged**